addons/mrp/models/stock_scrap.py

```diff
@@ -14,6 +13,12 @@ class StockScrap(models.Model):
         'mrp.workorder', 'Work Order',
         states={'done': [('readonly', True)]},
         check_company=True) # Not to restrict or prefer quants, but informative
+    product_is_kit = fields.Boolean(related='product_id.is_kits')
+    product_template = fields.Many2one(related='product_id.product_tmpl_id')
+    bom_id = fields.Many2one(
+        'mrp.bom', 'Kit',
+        domain="[('type', '=', 'phantom'), ('product_tmpl_id', '=', product_template)]",
```

Comment:
I would add a condition to ensure the BoM doesn't have a product define or the product define is the same than the scrap product